src_new/adapters/prebid.js

```diff
@@ -203,18 +252,43 @@ function generatedKeyCallback(adapterID, adUnits, adapterConfig, impressionID, g
 
 	var code, sizes, divID = currentSlot.getDivID();
 	
-	if(kgpConsistsWidthAndHeight){
-		code = refThis.getPBCodeWithWidthAndHeight(divID, adapterID, currentWidth, currentHeight);
-		sizes = [[currentWidth,currentHeight]];
-	}else{
-		code = refThis.getPBCodeWithoutWidthAndHeight(divID, adapterID);
-		sizes = currentSlot.getSizes();	
+	if(!CONFIG.getSingleImpressionSetting()){
+		if(kgpConsistsWidthAndHeight){
+			code = refThis.getPBCodeWithWidthAndHeight(divID, adapterID, currentWidth, currentHeight);
+			sizes = [[currentWidth,currentHeight]];
+		}else{
+			code = refThis.getPBCodeWithoutWidthAndHeight(divID, adapterID);
+			sizes = currentSlot.getSizes();	
+		}
+		refThis.kgpvMap [ code ] = {
+			kgpv: generatedKey,
+			divID: divID
+		};
+	} else{
+		// This will be executed in case single impression feature is enabled.
+		
+		code = currentSlot.getDivID();
+		sizes = currentSlot.getSizes();
+		if (refThis.kgpvMap [ code ] && refThis.kgpvMap [ code ].kgpvs.length > 0){
```

Comment:
please use the util function that we have for loop on an array, avoid duplication of code

Reply:
please add some comments above this code block about what it does